src/main/java/com/zendesk/maxwell/replication/BinlogConnectorReplicator.java

```diff
@@ -501,6 +503,10 @@ private RowMapBuffer getTransactionRows(BinlogConnectorEvent beginEvent) throws
 					String sql = qe.getSql();
 					String upperCaseSql = sql.toUpperCase();
 
+					if (sql.startsWith(BinlogConnectorEvent.XAEND)) {
+						LOGGER.info("XA END");
+						return buffer;
+					}
```

Comment:
move this into the else-if case plz.

Reply:
or... isn't this handled by `isCommitEvent`?  Can you explain?

Reply:
Just I think only _XA Commit_ and _XA ROLLBACK_ is the end of XA Transaction.